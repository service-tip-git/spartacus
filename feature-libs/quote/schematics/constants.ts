/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { SPARTACUS_QUOTE } from '@spartacus/schematics';

export const QUOTE_FOLDER_NAME = 'quote';
export const QUOTE_MODULE_NAME = 'quote';

export const QUOTE_SCSS_FILE_NAME = 'quote.scss';

export const SPARTACUS_QUOTE_ROOT = `${SPARTACUS_QUOTE}/root`;
export const SPARTACUS_QUOTE_ASSETS = `${SPARTACUS_QUOTE}/assets`;
export const QUOTE_TRANSLATIONS = 'quoteTranslations';
export const QUOTE_TRANSLATION_CHUNKS_CONFIG = 'quoteTranslationChunksConfig';
