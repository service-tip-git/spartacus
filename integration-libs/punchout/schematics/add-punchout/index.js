"use strict";
/*
 * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.addPunchoutFeatures = addPunchoutFeatures;
var schematics_1 = require("@angular-devkit/schematics");
var schematics_2 = require("@spartacus/schematics");
var package_json_1 = require("../../package.json");
function addPunchoutFeatures(options) {
    return function (tree, _context) {
        var packageJson = (0, schematics_2.readPackageJson)(tree);
        (0, schematics_2.validateSpartacusInstallation)(packageJson);
        var features = (0, schematics_2.analyzeCrossFeatureDependencies)(options.features);
        return (0, schematics_1.chain)([
            (0, schematics_2.analyzeApplication)(options, features),
            (0, schematics_2.addFeatures)(options, features),
            (0, schematics_2.addPackageJsonDependenciesForLibrary)(package_json_1.peerDependencies, options),
            (0, schematics_2.finalizeInstallation)(options, features),
        ]);
    };
}
//# sourceMappingURL=index.js.map