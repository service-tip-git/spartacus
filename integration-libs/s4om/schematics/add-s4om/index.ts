/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  chain,
  Rule,
  SchematicContext,
  Tree,
} from '@angular-devkit/schematics';
import {
  addFeatures,
  addPackageJsonDependenciesForLibrary,
  analyzeApplication,
  analyzeCrossFeatureDependencies,
  finalizeInstallation,
  LibraryOptions as SpartacusOrganizationOptions,
  readPackageJson,
  validateSpartacusInstallation,
} from '@spartacus/schematics';
import { peerDependencies } from '../../package.json';

export function addS4OMFeature(options: SpartacusOrganizationOptions): Rule {
  return (tree: Tree, _context: SchematicContext): Rule => {
    const packageJson = readPackageJson(tree);
    validateSpartacusInstallation(packageJson);

    const features = analyzeCrossFeatureDependencies(
      options.features as string[]
    );

    return chain([
      analyzeApplication(options, features),

      addFeatures(options, features),
      addPackageJsonDependenciesForLibrary(peerDependencies, options),

      finalizeInstallation(options, features),
    ]);
  };
}
