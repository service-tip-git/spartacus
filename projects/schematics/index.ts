/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

/*
 * Public API Surface of schematics
 */
export { Schema as SpartacusOptions } from './src/add-spartacus/schema';
export * from './src/shared/index';
