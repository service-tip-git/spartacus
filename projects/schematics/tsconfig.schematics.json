{
  "compilerOptions": {
    "baseUrl": "./",
    "module": "CommonJs",
    "types": ["node", "jest"],
    "paths": {
      "@spartacus/core": ["../../projects/core/public_api"]
    },
    "declaration": true,
    "noEmitOnError": true,
    "noFallthroughCasesInSwitch": true,
    "noImplicitAny": true,
    "noImplicitThis": true,
    "noUnusedParameters": true,
    "noUnusedLocals": true,
    "skipDefaultLibCheck": true,
    "skipLibCheck": true,
    "sourceMap": true,
    "strict": true,
    "strictNullChecks": true,
    "resolveJsonModule": true,
    "esModuleInterop": true
  },
  "include": ["index.ts", "src/**/*.ts"],
  "exclude": [
    "src/*/files/**/*",
    "src/**/*_spec.ts",
    "src/migrations/test/**/*"
  ]
}
