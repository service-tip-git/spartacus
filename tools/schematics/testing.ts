/*
 * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ChildProcess, exec, execSync } from 'child_process';
import { prompt } from 'enquirer';
import fs from 'fs';
import glob from 'glob';
import path from 'path';
import { chalk } from '../chalk';

const featureLibsFolders: string[] = [
  'asm',
  'cart',
  'order',
  'checkout',
  'organization',
  'pdf-invoices',
  'pickup-in-store',
  'product',
  'product-configurator',
  'product-multi-dimensional',
  'qualtrics',
  'requested-delivery-date',
  'estimated-delivery-date',
  'smartedit',
  'storefinder',
  'tracking',
  'user',
  'quote',
  'customer-ticketing',
];

const integrationLibsFolders: string[] = [
  'cdc',
  'cds',
  'digital-payments',
  'epd-visualization',
  's4om',
  'opf',
  'segment-refs',
  'opps',
  's4-service',
  'cpq-quote',
];

const commands = [
  'publish',
  'build projects/schematics',
  'build asm/schematics',
  'build cart/schematics',
  'build order/schematics',
  'build checkout/schematics',
  'build quote/schematics',
  'build cdc/schematics',
  'build s4-service/schematics',
  'build cds/schematics',
  'build digital-payments/schematics',
  'build epd-visualization/schematics',
  'build opf/schematics',
  'build organization/schematics',
  'build pdf-invoices/schematics',
  'build pickup-in-store/schematics',
  'build product/schematics',
  'build product-configurator/schematics',
  'build product-multi-dimensional/schematics',
  'build s4om/schematics',
  'build segment-refs/schematics',
  'build opps/schematics',
  'build qualtrics/schematics',
  'build requested-delivery-date/schematics',
  'build estimated-delivery-date/schematics',
  'build smartedit/schematics',
  'build storefinder/schematics',
  'build tracking/schematics',
  'build user/schematics',
  'build customer-ticketing/schematics',
  'build cpq-quote/schematics',
  'build all libs',
  'test all schematics',
  'exit',
] as const;
type Command = (typeof commands)[number];

const buildLibRegEx = new RegExp('build (.*?)/schematics');
const verdaccioRegistryUrl = 'http://localhost:4873/';
const originalRegistryUrl = execSync('npm config get @spartacus:registry')
  .toString()
  .trim();

function startVerdaccio(): ChildProcess {
  execSync('rm -rf ./scripts/install/storage');

  console.log('Waiting for verdaccio to boot...');
  const res = exec('verdaccio --config ./scripts/install/config.yaml');
  try {
    execSync(`npx wait-on ${verdaccioRegistryUrl} --timeout 10000`);
  } catch (_e) {
    console.log(
      chalk.red(
        `\n‚ùå Couldn't boot verdaccio. Make sure to install it globally: \n> npm i -g verdaccio@4`
      )
    );
    process.exit(1);
  }
  console.log('Pointing npm to verdaccio');
  execSync(`npm config set @spartacus:registry ${verdaccioRegistryUrl}`);
  return res;
}

function beforeExit(): void {
  console.log('Setting npm back to npmjs.org');
  execSync(`npm config set @spartacus:registry ${originalRegistryUrl}`);
  if (verdaccioProcess) {
    try {
      console.log('Killing verdaccio');
      verdaccioProcess.kill();
    } catch {}
  }
}

/**
 * Object describing the result of publishing a package
 */
type PackagePublishingResult = {
  packageName: string;
  stdout: string;
  stderr: string;
  error?: any;
};

/**
 * Get list of package.json files to publish
 */
function getPackageJsonFiles(): string[] {
  const sourceFiles = [
    'projects/storefrontstyles/package.json',
    'projects/schematics/package.json',
  ];
  const distFiles = glob.sync(`dist/!(node_modules)/package.json`);
  return [...sourceFiles, ...distFiles];
}

/**
 * Publish a single package to Verdaccio
 */
function publishPackage(packagePath: string): Promise<PackagePublishingResult> {
  return new Promise((resolve, reject) => {
    const packageJsonContent = JSON.parse(
      fs.readFileSync(packagePath, 'utf-8')
    );
    const directory = path.dirname(packagePath);
    const command = `cd ${directory} && npm publish --registry=${verdaccioRegistryUrl} --no-git-tag-version --color always`;
    exec(command, {}, (error, stdout, stderr) => {
      if (error) {
        reject({
          packageName: packageJsonContent.name,
          stdout,
          stderr,
          error: `Command failed: ${error.cmd}`,
        });
      } else {
        resolve({
          packageName: packageJsonContent.name,
          stdout,
          stderr,
        });
      }
    });
  });
}

/**
 * Display final results of publishing packages
 */
function printAllPackagesPublishingResults({
  successful,
  failed,
}: {
  successful: PackagePublishingResult[];
  failed: PackagePublishingResult[];
}): void {
  console.log('\n=== Publishing All Packages Results ===');

  if (successful.length > 0) {
    console.log(
      chalk.green(`\n‚úÖ Successfully published ${successful.length} packages:`)
    );
    successful.forEach(printPackagePublishingResult);
  }

  if (failed.length > 0) {
    console.log(chalk.red(`\n‚ùå Failed to publish ${failed.length} packages:`));
    failed.forEach(printPackagePublishingResult);
    console.log(
      chalk.red(
        `\nüö® ${failed.length} packages could not be published. Please check the errors and logs above.`
      )
    );
    throw new Error(); // reject the result promise
  } else {
    console.log(chalk.green('\nüéâ All packages published successfully!'));
  }
}

/**
 * Show a result of publishing a single package
 *
 * @example
 * ================================
 * ‚ùå/‚úÖ package name
 * ================================
 * stdout
 * stderr
 * (ERROR: error message)
 */
function printPackagePublishingResult(result: PackagePublishingResult): void {
  const separator = '='.repeat(50);
  const icon = result.error ? '‚ùå' : '‚úÖ';
  const chalkColor = result.error ? chalk.red : chalk.green;
  console.log(
    chalkColor(
      chalk.bold(`\n${separator}\n${icon} ${result.packageName}\n${separator}`)
    )
  );

  if (result.stdout) {
    console.log(result.stdout);
  }
  if (result.stderr) {
    console.log(result.stderr);
  }
  if (result.error) {
    console.error(chalk.red(`ERROR:\n${result.error}`));
  }
}

/**
 * Show progress of publishing packages
 *
 * @example
 * [1/n] ‚ùå/‚úÖ package name 1
 * [2/n] ‚ùå/‚úÖ package name 2
 * ...
 */
function printPackagesPublishingProgress(
  result: PackagePublishingResult,
  current: number,
  total: number
): void {
  const icon = result.error ? '‚ùå' : '‚úÖ';
  console.log(`\n[${current}/${total}] ${icon} ${result.packageName}`);
}

/**
 * Publish all packages to Verdaccio
 */
async function publishAllPackages(): Promise<void> {
  //Since migration to NPM, packages will be published with version that has been set in package.json files
  //and before each subsequent release Verdaccio will be cleaned up so that there is no conflict due to the version of existing packages.
  //Because of the strategy used by NPM to resolve peer dependencies, any upgrade of a package's version would also require upgrading the version of that package in all places where it is used as a peer dependency.
  //This in turn would cause too much noise

  //clear Verdaccio storage
  execSync('rm -rf ./scripts/install/storage', { stdio: 'ignore' });

  const allFiles = getPackageJsonFiles();
  console.log(
    chalk.green(
      `\n‚è≥ Publishing ${allFiles.length} packages in parallel (please wait)...\n`
    )
  );

  /** Number of packages published, successfully or not */
  let completedCount = 0;

  // Run all publish operations in parallel and display progress
  const results = await Promise.allSettled(
    allFiles.map((packagePath) =>
      publishPackage(packagePath).then((result) => {
        printPackagesPublishingProgress(
          result,
          ++completedCount,
          allFiles.length
        );
        return result;
      })
    )
  );

  const successful = results
    .filter((result) => result.status === 'fulfilled')
    .map((result) => result.value);
  const failed = results
    .filter((result) => result.status === 'rejected')
    .map((result) => result.reason);

  printAllPackagesPublishingResults({ successful, failed });
}

function buildLibs(): void {
  execSync('npm run build:libs', { stdio: 'inherit' });
}

async function buildSchematics(
  options: { publish: boolean } = { publish: false }
): Promise<void> {
  execSync('npm run build:schematics', { stdio: 'inherit' });
  if (options.publish) {
    await publishAllPackages();
  }
}

async function buildSchematicsAndPublish(buildCmd: string): Promise<void> {
  await buildSchematics();
  execSync(buildCmd, {
    stdio: 'inherit',
  });
  await publishAllPackages();
}

function testAllSchematics(): void {
  try {
    execSync('npm --prefix projects/schematics run test --coverage', {
      stdio: 'inherit',
    });

    featureLibsFolders.forEach((lib) =>
      execSync(
        `npm --prefix feature-libs/${lib} run test:schematics --coverage`,
        {
          stdio: 'inherit',
        }
      )
    );
    integrationLibsFolders.forEach((lib) =>
      execSync(
        `npm --prefix integration-libs/${lib} run test:schematics --coverage`,
        {
          stdio: 'inherit',
        }
      )
    );
  } catch (e) {
    console.error(e);
    beforeExit();
    process.exit();
  }
}

const isVoiceNotifyEnabled = process.argv.includes('--voice-notify');
if (isVoiceNotifyEnabled) {
  console.log('Voice notifications enabled');
}

/**
 * Notify the user by voice message
 */
function notify(message: string): void {
  if (!isVoiceNotifyEnabled) return;
  try {
    execSync(`say "${message}"`);
  } catch (error) {
    console.warn('Voice notification failed:', error);
  }
}

/**
 * Run task and notify the user when it's completed
 */
async function notifyOnComplete<T>({
  taskName,
  task,
}: {
  taskName: string;
  task: (() => T) | (() => Promise<T>);
}): Promise<unknown> {
  try {
    const maybePromiseResult = task();
    const result = await (maybePromiseResult instanceof Promise
      ? maybePromiseResult
      : Promise.resolve(maybePromiseResult));
    notify(`${taskName} completed`);
    return result;
  } catch (error) {
    notify(`${taskName} failed`);
    return error;
  }
}

async function executeCommand(command: Command): Promise<void> {
  switch (command) {
    case 'publish':
      await notifyOnComplete({
        taskName: 'Publishing',
        task: () => publishAllPackages(),
      });
      break;
    case 'build projects/schematics':
      await notifyOnComplete({
        taskName: 'Schematics build',
        task: () => buildSchematics({ publish: true }),
      });
      break;
    case 'build asm/schematics':
    case 'build cart/schematics':
    case 'build order/schematics':
    case 'build checkout/schematics':
    case 'build quote/schematics':
    case 'build cpq-quote/schematics':
    case 'build cdc/schematics':
    case 'build s4-service/schematics':
    case 'build cds/schematics':
    case 'build digital-payments/schematics':
    case 'build epd-visualization/schematics':
    case 'build opf/schematics':
    case 'build organization/schematics':
    case 'build pdf-invoices/schematics':
    case 'build pickup-in-store/schematics':
    case 'build product/schematics':
    case 'build product-configurator/schematics':
    case 'build product-multi-dimensional/schematics':
    case 'build qualtrics/schematics':
    case 'build requested-delivery-date/schematics':
    case 'build estimated-delivery-date/schematics':
    case 'build s4om/schematics':
    case 'build segment-refs/schematics':
    case 'build opps/schematics':
    case 'build smartedit/schematics':
    case 'build storefinder/schematics':
    case 'build tracking/schematics':
    case 'build user/schematics':
    case 'build customer-ticketing/schematics':
      const lib =
        buildLibRegEx.exec(command)?.pop() ?? 'LIB-REGEX-DOES-NOT-MATCH';
      await notifyOnComplete({
        taskName: `Schematics of ${lib} build`,
        task: () => buildSchematicsAndPublish(`npm run build:${lib}`),
      });
      break;
    case 'build all libs':
      await notifyOnComplete({
        taskName: 'All libraries build',
        task: () => buildLibs(),
      });
      break;
    case 'test all schematics':
      await notifyOnComplete({
        taskName: 'All schematics tests',
        task: () => testAllSchematics(),
      });
      break;
    case 'exit':
      beforeExit();
      process.exit();
  }
}

let verdaccioProcess: ChildProcess | undefined;

async function program(): Promise<void> {
  verdaccioProcess = startVerdaccio();
  try {
    while (true) {
      const response: { command: Command } = await prompt({
        name: 'command',
        type: 'select',
        message: 'What do you want to do next?',
        choices: [...commands],
      });

      await executeCommand(response.command);
    }
  } catch (e) {
    console.error(e);
    beforeExit();
    process.exit();
  }
}

// Handle killing the script
process.once('SIGINT', function () {
  beforeExit();
  process.exit();
});

process.on('SIGHUP', function () {
  beforeExit();
  process.exit();
});

process.once('SIGTERM', function () {
  beforeExit();
  process.exit();
});

program();
