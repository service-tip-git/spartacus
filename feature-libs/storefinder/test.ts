/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

// This file is required by karma.conf.js and loads recursively all the .spec and framework files

// Patching Object.defineProperty unlocks frozen JS symbols and makes possible to mock them.
// Should be used with caution, and only if there is no other way to mock stuff (eg. by DI)
// Has to be imported just after zone.js imports.
import 'testing/patch-object-define-property';

import { getTestBed } from '@angular/core/testing';
import {
  BrowserDynamicTestingModule,
  platformBrowserDynamicTesting,
} from '@angular/platform-browser-dynamic/testing';

// First, initialize the Angular testing environment.
getTestBed().initTestEnvironment(
  BrowserDynamicTestingModule,
  platformBrowserDynamicTesting(),
  {
    teardown: { destroyAfterEach: false },
  }
);
