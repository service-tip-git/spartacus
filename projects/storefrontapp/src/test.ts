/*
 * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
 *
 * SPDX-License-Identifier: Apache-2.0
 */

/* eslint-disable */
// This file is required by karma.conf.js and loads recursively all the .spec and framework files

import 'zone.js/dist/long-stack-trace-zone';
import 'zone.js/dist/proxy.js';
import 'zone.js/dist/sync-test';
import 'zone.js/dist/jasmine-patch';
import 'zone.js/dist/async-test';
import 'zone.js/dist/fake-async-test';
import { getTestBed } from '@angular/core/testing';
import {
  BrowserDynamicTestingModule,
  platformBrowserDynamicTesting,
} from '@angular/platform-browser-dynamic/testing';

// Unfortunately there's no typing for the `__karma__` variable. Just declare it as any.
declare var __karma__: any;

// Prevent Karma from running prematurely.
__karma__.loaded = function () {
  // Intentional empty function
};

// First, initialize the Angular testing environment.
getTestBed().initTestEnvironment(
  BrowserDynamicTestingModule,
  platformBrowserDynamicTesting(),
  {
    teardown: { destroyAfterEach: false },
  }
);
// Finally, start Karma to run the tests.
__karma__.start();
