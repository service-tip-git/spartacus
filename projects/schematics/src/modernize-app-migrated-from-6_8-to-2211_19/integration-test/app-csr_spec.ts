import { TempScopedNodeJsSyncHost } from '@angular-devkit/core/node/testing';
import { HostTree } from '@angular-devkit/schematics';
import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import { join } from 'path';
import { loadFixturesIntoTree } from '../../test/fixtures-test-utils';

const collectionPath = join(__dirname, '../../collection.json');

const SCHEMATICS_NAME = 'modernize-app-migrated-from-6_8-to-2211_19';

describe(`Schematics "${SCHEMATICS_NAME}" in CSR app`, () => {
  let host: TempScopedNodeJsSyncHost;
  let tree: UnitTestTree;
  let runner: SchematicTestRunner;

  beforeAll(async () => {
    host = new TempScopedNodeJsSyncHost();
    tree = new UnitTestTree(new HostTree(host));
    runner = new SchematicTestRunner('test', collectionPath);

    loadFixturesIntoTree({
      tree,
      fixturesDir: join(__dirname, '__fixtures__/app-csr'),
    });

    tree = await runner.runSchematic(SCHEMATICS_NAME, {}, tree);
  });

  it('should update the angular.json', () => {
    expect(tree.readContent('/angular.json')).toMatchSnapshot();
  });

  it('should update the tsconfig.json', () => {
    expect(tree.readContent('/tsconfig.json')).toMatchSnapshot();
  });
});
