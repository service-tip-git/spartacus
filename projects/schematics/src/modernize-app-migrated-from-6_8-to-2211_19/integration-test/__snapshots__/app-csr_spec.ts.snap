// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schematics "modernize-app-migrated-from-6_8-to-2211_19" in CSR app should update the angular.json 1`] = `
"{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "test-app-migrated-from-6_8-to-2211_19-csr": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "scss"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:application",
          "options": {
            "outputPath": "dist/test-app-migrated-from-6_8-to-2211_19-csr",
            "index": "src/index.html",
            "polyfills": [
              "zone.js"
            ],
            "tsConfig": "tsconfig.app.json",
            "inlineStyleLanguage": "scss",
            "assets": [
              "src/favicon.ico",
              "src/assets",
              {
                "glob": "**/*",
                "input": "./node_modules/@spartacus/smartedit/assets",
                "output": "assets/"
              }
            ],
            "styles": [
              "src/styles.scss",
              "src/styles/spartacus/user.scss",
              "src/styles/spartacus/cart.scss",
              "src/styles/spartacus/order.scss",
              "src/styles/spartacus/checkout.scss",
              "src/styles/spartacus/storefinder.scss",
              "src/styles/spartacus/asm.scss",
              "src/styles/spartacus/product.scss"
            ],
            "scripts": [],
            "stylePreprocessorOptions": {
              "includePaths": [
                "node_modules/"
              ]
            },
            "browser": "src/main.ts"
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "2.5mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "optimization": false,
              "extractLicenses": false,
              "sourceMap": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "buildTarget": "test-app-migrated-from-6_8-to-2211_19-csr:build:production"
            },
            "development": {
              "buildTarget": "test-app-migrated-from-6_8-to-2211_19-csr:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "buildTarget": "test-app-migrated-from-6_8-to-2211_19-csr:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "polyfills": [
              "zone.js",
              "zone.js/testing"
            ],
            "tsConfig": "tsconfig.spec.json",
            "inlineStyleLanguage": "scss",
            "assets": [
              "src/favicon.ico",
              "src/assets",
              {
                "glob": "**/*",
                "input": "./node_modules/@spartacus/smartedit/assets",
                "output": "assets/"
              }
            ],
            "styles": [
              "src/styles.scss",
              "src/styles/spartacus/user.scss",
              "src/styles/spartacus/cart.scss",
              "src/styles/spartacus/order.scss",
              "src/styles/spartacus/checkout.scss",
              "src/styles/spartacus/storefinder.scss",
              "src/styles/spartacus/asm.scss",
              "src/styles/spartacus/product.scss"
            ],
            "scripts": [],
            "stylePreprocessorOptions": {
              "includePaths": [
                "node_modules/"
              ]
            }
          }
        }
      }
    }
  }
}"
`;

exports[`Schematics "modernize-app-migrated-from-6_8-to-2211_19" in CSR app should update the tsconfig.json 1`] = `
"{
  "compileOnSave": false,
  "compilerOptions": {
    "outDir": "./dist/out-tsc",
    "strict": true,
    "noImplicitOverride": true,
    "noPropertyAccessFromIndexSignature": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "sourceMap": true,
    "declaration": false,
    "experimentalDecorators": true,
    "moduleResolution": "node",
    "importHelpers": true,
    "target": "ES2022",
    "module": "ES2022",
    "useDefineForClassFields": false,
    "lib": [
      "ES2022",
      "dom"
    ],
    "skipLibCheck": true,
    "esModuleInterop": true
  },
  "angularCompilerOptions": {
    "enableI18nLegacyMessageIdFormat": false,
    "strictInjectionParameters": true,
    "strictInputAccessModifiers": true,
    "strictTemplates": true
  }
}"
`;
